docs/window-shims-and-polyfills.md

```diff
@@ -0,0 +1,120 @@
+# Window shims and polyfills.
+
+Please consult the [gotchas](./gotchas.md), as it contains the preferred way to solve the problems where we need those shims and polyfills for.
+
+As part of the default installation we now provide a sample file on how you can provide `window` and `document` shims to your universal application. This file serves as a starting point.
+
+> ## Note
+> Please be aware that using the global scope can lead to memory leaks and security issues.
+> The shims and polyfills provided will not cause this by themselves, but if your application or any of its dependencies
+> stores or modifies _anything_ that is in the global scope (this includes, but is not limited to `window` and `document`) you are at risk.
+
+
+## Why
+
+We see a lot of questions on ngUniversal that revolve around using 3rd party libraries. With this addition, we are providing a starting point to give you some samples and an idea on how to tackle issues related to using 3rd party libraries. We would like to emphasize that these techniques are a last-resort solution. The preferred way to deal with 3rd party libraries is by utilizing the Angular injector.
+
+## Location
+
+We have generated a file called `windows-shims-and-polyfiles.ts` in a subfolder of the location where `server.ts is` created.
+
+## How
+
+Because we need to compile your application's code for use in node, we need 3 types of helpers:
+1. TypeScript shims
+2. Node shims
+3. node polyfills
+
+Those are needed even if the code isn't used when your app is rendered during SSR as Universal still needs to read and compile all of the code in your application.
+
+### 1. TypeScript shims
+
```

Comment:
"when you look into the provided shims and polyfills file

Reply:
@joe, where you want this?